test/utils.ts

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { PROCESSOR_TYPE, WORKFLOW_TYPE } from '../common/constants';
+import {
+  IProcessorConfig,
+  ProcessorsConfig,
+  Workflow,
+} from '../common/interfaces';
+
+export function mockStore(
+  workflowId: string,
+  workflowName: string,
+  workflowType: WORKFLOW_TYPE
+) {
+  return {
+    getState: () => ({
+      opensearch: {
+        errorMessage: '',
+      },
+      ml: {},
+      workflows: {
+        loading: false,
+        errorMessage: '',
+        workflows: {
+          [workflowId]: generateWorkflow(
+            workflowId,
+            workflowName,
+            workflowType
+          ),
+        },
+      },
+    }),
+    dispatch: jest.fn(),
+    subscribe: jest.fn(),
+    replaceReducer: jest.fn(),
+    [Symbol.observable]: jest.fn(),
+  };
+}
+
+function generateWorkflow(
+  workflowId: string,
+  workflowName: string,
+  workflowType: WORKFLOW_TYPE
+): Workflow {
+  return {
+    id: workflowId,
+    name: workflowName,
+    version: { template: '1.0.0', compatibility: ['2.17.0', '3.0.0'] },
+    ui_metadata: {
```

Comment:
The schema & default values inside of `ui_metadata` are still changing, and will definitely change more in the future. For consistency and maintainability, how about we leverage the helper fns here? https://github.com/opensearch-project/dashboards-flow-framework/blob/main/public/pages/workflows/new_workflow/utils.ts

This is the source of truth for the default configs. If we reuse it in tests, we can guarantee the tests and UI would be consuming the same values.

Still worth having the params like id/name/type here. Maybe can use the existing fns for generating all of the metadata, and then overwrite a few things like id/name based on the params.

Reply:
It's my bad for not showing you that helper fn before. Hopefully that makes more sense and gives you some more understanding on how we generate the configs in general :)